application/common/models/Method.php

```diff
@@ -61,4 +93,159 @@ public function getRawPhoneNumber()
     {
         return preg_replace('/,/', '', $this->value);
     }
+
+    /**
+     * @param integer $userId
+     * @param string $type
+     * @param string $value
+     * @return Method
+     * @throws \Exception
+     */
+    public static function createAndSendVerification($userId, $type, $value)
+    {
+        $log = [
+            'class' => __CLASS__,
+            'method' => 'createAndSendVerification',
```

Comment:
I think you can use `__METHOD__` here if you don't want to retype the method name for these sorts of log entries. (http://php.net/manual/en/language.constants.predefined.php)